### PR TITLE
[TDV-147] Integrate account role resetting

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/vue/pages/Verification.vue [hunk 1]
```diff
@@ -155,7 +172,7 @@ export default {
   &--rejected { @include apply-theme($col-request-rejected) }
 }
 
-.kyc-request-state__content {
+.verification__state-message__content {
```
Comment (max-nes) [inline src/vue/pages/Verification.vue]: invalid BEM

File: src/js/records/requests/change-role.record.js [hunk 2]
```diff
@@ -8,13 +8,17 @@ export class ChangeRoleRequestRecord {
     this.requestor = _get(record, 'requestor.id')
     this.reviewer = _get(record, 'reviewer.id')
     this.reference = record.reference
-    this.rejectReason = record.rejectReason
+    this.rejectReason = record.rejectReason ||
+      _get(record, 'requestDetails.creatorDetails.reason')
```
Comment (max-nes) [inline src/js/records/requests/change-role.record.js]: is not reject reason actually

Comment (ardier16) [reply src/js/records/requests/change-role.record.js]: Extracted to the separate field

File: src/vue/pages/Verification.vue (hunk 1 above)
Comment (ardier16) [reply src/vue/pages/Verification.vue]: Fixed

File: src/vue/pages/Verification.vue [hunk 3]
```diff
@@ -1,26 +1,42 @@
 <template>
   <div class="verification">
     <div
-      v-if="kycState === REQUEST_STATES_STR.approved"
-      class="kyc-request-state kyc-request-state--approved"
+      v-if="isAccountRoleReset"
```
Comment (max-nes) [inline src/vue/pages/Verification.vue]: reseted

File: src/vue/pages/Verification.vue [hunk 4]
```diff
@@ -122,6 +138,8 @@ export default {
     ...mapGetters({
       kycState: vuexTypes.kycState,
       kycRejectReason: vuexTypes.kycRequestRejectReason,
+      kycResetReason: vuexTypes.kycRequestResetReason,
+      isAccountRoleReset: vuexTypes.isAccountRoleReset,
```
Comment (max-nes) [inline src/vue/pages/Verification.vue]: reseted

File: src/vue/pages/VerificationCorporate.vue [hunk 5]
```diff
@@ -210,7 +210,17 @@ export default {
...
       kvEntryCorporateRoleId: vuexTypes.kvEntryCorporateRoleId,
+      isAccountRoleReset: vuexTypes.isAccountRoleReset,
+      previousAccountRole: vuexTypes.kycPreviousAccountRole,
     }),
+    isFormEditable () {
+      return this.isAccountRoleReset ||
+        this.kycState === REQUEST_STATES_STR.rejected
+    },
+    isFormAutocompleted () {
+      return (this.previousAccountRole === this.kvEntryCorporateRoleId) ||
```
Comment (max-nes) [inline src/vue/pages/VerificationCorporate.vue]: what about general

File: src/vue/pages/VerificationCorporate.vue [hunk 6]
```diff
@@ -210,7 +210,17 @@ export default {
...
+        (!this.isAccountRoleReset && this.kycState)
```
Comment (max-nes) [inline src/vue/pages/VerificationCorporate.vue]: move to another computed, cause idk what this is about

File: src/vue/pages/VerificationCorporate.vue [hunk 7]
```diff
@@ -210,7 +210,17 @@ export default {
...
     ...mapGetters({
       kvEntryCorporateRoleId: vuexTypes.kvEntryCorporateRoleId,
+      isAccountRoleReset: vuexTypes.isAccountRoleReset,
+      previousAccountRole: vuexTypes.kycPreviousAccountRole,
     }),
+    isFormEditable () {
+      return this.isAccountRoleReset ||
+        this.kycState === REQUEST_STATES_STR.rejected
+    },
+    isFormAutocompleted () {
```
Comment (max-nes) [inline src/vue/pages/VerificationCorporate.vue]: isFormPopulatable

File: src/vue/pages/Verification.vue (hunk 3 above)
Comment (ardier16) [reply src/vue/pages/Verification.vue]: Renamed

File: src/vue/pages/Verification.vue (hunk 4 above)
Comment (ardier16) [reply src/vue/pages/Verification.vue]: Renamed

File: src/vue/pages/VerificationCorporate.vue (hunk 5 above)
Comment (ardier16) [reply src/vue/pages/VerificationCorporate.vue]: It is handled in general form, we don't need to fill the corporate form in that case

File: src/vue/pages/VerificationCorporate.vue (hunk 6 above)
Comment (ardier16) [reply src/vue/pages/VerificationCorporate.vue]: Refactored a bit

File: src/vue/pages/VerificationCorporate.vue (hunk 7 above)
Comment (ardier16) [reply src/vue/pages/VerificationCorporate.vue]: Renamed

**Pull request merged**